dixit/src/main/java/tw/wally/dixit/model/Round.java

```diff
@@ -0,0 +1,140 @@
+        }
+        if (!storyteller.equals(story.getPlayer())) {
+            throw new IllegalArgumentException("Guesser can't tell the story.");
+        }
+        this.story = story;
+        roundState = RoundState.CARD_PLAYING;
+    }
+
+    public void addPlayCard(PlayCard playCard) {
+        if (RoundState.CARD_PLAYING != roundState) {
+            throw new IllegalArgumentException("When the round state isn't card playing, guesser can't play the card.");
+        }
+        if (playCards.size() == numberOfGuessers) {
+            throw new IllegalArgumentException(format("Number of playCards can't be higher than %d.", numberOfGuessers));
+        }
+        var player = playCard.getPlayer();
+        if (contains(playCards.values(), card -> card.getPlayer().equals(player))) {
+            throw new IllegalArgumentException(format("Player: %s can't play the card twice in same round.", player.getName()));
+        }
+        playCards.put(playCard.getCardId(), playCard);
+        if (playCards.size() == numberOfGuessers) {
+            roundState = RoundState.PLAYER_GUESSING;
+        }
+    }
+
+    public void addGuess(Guess guess) {
+        if (RoundState.PLAYER_GUESSING != roundState) {
+            throw new IllegalArgumentException("When the round state isn't player guessing, guesser can't guess the story.");
+        }
+        if (guesses.size() == numberOfGuessers) {
+            throw new IllegalArgumentException(format("Number of guesses can't be higher than %d.", numberOfGuessers));
+        }
+        var guesser = guess.getGuesser();
+        if (guesses.containsKey(guesser)) {
+            throw new IllegalArgumentException(format("Guesser: %s can't guess the story twice in same round.", guesser.getName()));
+        }
+        guesses.put(guesser, guess);
+        if (guesses.size() == numberOfGuessers) {
+            roundState = RoundState.SCORING;
+        }
+    }
+
+    public void score() {
+        if (RoundState.SCORING != roundState) {
+            throw new IllegalArgumentException("When the round state isn't scoring, Round can't score.");
+        }
+        long numberOfAllGuessersWhoGuessesStoryCorrectly =
+                count(guesses.values(), guess -> storyteller.equals(guess.getPlayCard().getPlayer()));
+        if (numberOfAllGuessersWhoGuessesStoryCorrectly == numberOfGuessers) {
+            guessers.forEach(player -> player.addScore(2));
+        } else if (numberOfAllGuessersWhoGuessesStoryCorrectly == 0) {
+            guessers.forEach(player -> player.addScore(2));
+            addBonusScoreToGuesserWhoseCardIsGuessedByOtherGuesser();
+        } else if (numberOfAllGuessersWhoGuessesStoryCorrectly < numberOfGuessers) {
+            storyteller.addScore(ANSWER_CORRECTLY_SCORE);
+            addScoreToGuesserWhoGuessedStoryCorrectly();
+            addBonusScoreToGuesserWhoseCardIsGuessedByOtherGuesser();
```

Comment:
The code is not clean enough, as the `Round` class is not modeled perfectly; 
you can improve it by Introducing a new class `Guesser(player: Player, guess?: Guess, playCard?: PlayCard)` 

Then you can refactor your `Round` class into

```java
class Round {
    private final Player storyteller;
    private final List<Guesser> guessers;
    private RoundState roundState;
    private Story story;
    ....
```

Isn't this very clean?